vizro-core/src/vizro/models/_components/ag_grid.py

```diff
@@ -126,7 +120,7 @@ def __call__(self, **kwargs):
             kwargs["data_frame"] = data_manager[self["data_frame"]].load()
         figure = self.figure(**kwargs)
         figure.id = self._inner_component_id
-        return figure
+        return html.Div([figure, dcc.Store(id=f"{self._inner_component_id}_guard_actions_chain", data=True)])
```

Comment:
Key change: we return `*_guard_actions_chain` to operate the `guard_actions_chain` callback mechanism.